<a href="https://colab.research.google.com/github/gurram46/gurram46-Estimation-And-Confidence-Intervals/blob/main/Estimation_And_Confidence_Intervals_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

This is a Scenaro based assignemet

# Background :
In quality control processes, especially when dealing with high-value items, destructive sampling is a necessary but costly method to ensure product quality. The test to determine whether an item meets the quality standards destroys the item, leading to the requirement of small sample sizes due to cost constraints.

**Understanding :**
In quality control, destructive sampling is costly but necessary. Since testing destroys the item, only small samples can be tested to control costs.

# Scenario:
A manufacturer of print-heads for personal computers is interested in estimating the mean durability of their print-heads in terms of the number of characters printed before failure. To assess this, the manufacturer conducts a study on a small sample of print-heads due to the destructive nature of the testing process.

**Understanding :**
A manufacturer wants to estimate the mean durability of print-heads (measured in millions of characters printed before failure) through destructive testing. They tested 15 randomly selected print-heads


# Data:
A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:
1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29

**Understanding :**
This dataset will be used to analyze the mean durability of the print-heads and estimate the population mean through confidence intervals. Given the small sample size, the study needs to account for the sample standard deviation or, if available, the known population standard deviation to ensure accurate interval estimation

# Assignment Tasks

# Task-1
**Build 99% Confidence Interval Using Sample Standard Deviation**

Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.



**Goal**: Estimate the mean durability using the sample standard deviation.

**Why t-Distribution?**: Small sample size (n = 15) and unknown population standard deviation require t-distribution for accuracy

In [2]:
import numpy as np
from scipy import stats



*   we are using numpy to perform efficient numerical operations such as  calculating mean and standard deviation

*   To access statistical functions like t-distribution and z-distribution critical values

In [3]:
data= [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
n=len(data)



*   Here we are loading the data which is given creating a list name is data
*   And n is a variable we're using to store the number of data points in the dataset



In [4]:
sample_mean = np.mean(data)
sample_mean

1.2386666666666666



*   Here we calucating the mean of the data using numpay

*   The mean of the data is 1.2386666666666666



In [5]:
sample_std = np.std(data, ddof=1)
sample_std

0.19316412956959936



*   Now we are using numpy for sample standard devation

*   The sample standard devation for data is 0.19316412956959936



In [6]:
df = n - 1
df

14

**Degree of freedom :**

Dof is defined as the number of independent parameters required to specify the locationof every link with in the mechanism completely.

dof=n-1

In [7]:
t_value = stats.t.ppf(0.995, df)
t_value

2.976842734370834




*   Here we find the Critical t-Value for 99% Confidence Level
*   99% confidence level corresponds to 0.995 for two-tailed
*   The value of t_value is 2.9768







In [10]:
margin_of_error=t_value * (sample_std / np.sqrt(n))
margin_of_error

0.14846932822817596



*   Here we are calculating margin of error by using numpy
*   The value of margin of errors is 0.14846



In [16]:
confidence_interval= (sample_mean-margin_of_error,sample_mean+margin_of_error)
confidence_interval

(1.0901973384384906, 1.3871359948948425)

# Confidence interval:
we are calicating the confdence interval using sample maen,margin of error

The the value of Confidence interval is lie between 1.09019 to 1.38713

In [17]:
print(f"99% Confidence interval (Using Sample Standard Deviation):{confidence_interval}")

99% Confidence interval (Using Sample Standard Deviation):(1.0901973384384906, 1.3871359948948425)


This confidence interval tells you that, with 99% confidence, the true mean number of characters printed before the print-head fails lies between 1.090 and 1.387 million characters.

# Task-2
 **Build 99% Confidence Interval Using Known Population Standard Deviation**

 If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.

 Goal:

 To estimate the range (confidence interval) in which the true mean durability of print-heads lies, with 99% confidence, using the known population standard deviation

In [18]:
population_std=0.2

In task to first we have to load the data but we have used the data ready in the Task-1 so there is no need to load the data and there is no need to import numpy and stats from scipy and The population standard deviation is alrady provided which is o.2

In [19]:
sample_mean = np.mean(data)
sample_mean

1.2386666666666666

So we are calucalating the sample mean which is 1.23866..

In [22]:
z_value = stats.norm.ppf(0.995)
z_value

2.5758293035489004



*   Here we find the Critical z-Value for 99% Confidence Level
*   99% confidence level corresponds to 0.995 for two-tailed
*   The value of z_value is 2.5758





In [23]:
margin_of_error = z_value * (population_std/np.sqrt(n))
margin_of_error

0.13301525327090588

We are calucating the margin_of_error by using z_value,population_std and sqrt(n)

In [24]:
confidence_interval = (sample_mean - margin_of_error, sample_mean + margin_of_error)
confidence_interval

(1.1056514133957607, 1.3716819199375725)

we are calicating the confdence interval using sample maen,margin of error

The the value of Confidence interval is lie between 1.105651 to 1.3716819

In [25]:
print(f"99% confidence_interval (using known population standrad devation): {confidence_interval}")

99% confidence_interval (using known population standrad devation): (1.1056514133957607, 1.3716819199375725)


This confidence interval tells you that, with 99% confidence, the true mean number of characters printed before the print-head fails lies between 1.204 and 1.396 million characters